sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -37,6 +37,16 @@ import org.apache.spark.sql.rapids.{CudfAggregate, GpuAggregateExpression, GpuDe
 import org.apache.spark.sql.types.{DoubleType, FloatType}
 import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
 
+object AggregateUtils {
+  /** Return true if the Attribute passed is one of aggregates in the list */
+  def validateAggregate(agg: AttributeSet): Boolean = {
+    val distinctAggs = List("avg", "sum", "count")
+    val res = agg.toSeq.exists(
+      agg => distinctAggs.exists(distinctAgg => agg.name.contains(distinctAgg)))
+    res
```

Comment:
>  The main thing is that we document exactly what we are doing and exactly what each of the functions are doing so no one accidentally uses one of the fuzzy match functions in error and so when we finally fix it for real we know exactly what to rip out again.
> 
+1

Reply:
While testing, I came across the bug where it fails if `non-distinct` function is not among "avg", "sum" or "count".
`partial_first` is actually included for `non-distinct` aggregations. So we need to include all aggregations in `distinctAggs` :( (need to rename variable) .

With current `distinctAggs` list, it would fail if I have a query something like this:
```
 SELECT
   first(ints),
   count(distinct doubles),
   count(distinct longs)
 FROM testTable
   group by longs
```
From the plan, we see that it will not fall back to CPU  since `first` is not matched. 

```
   +- *(2) HashAggregate(keys=[longs#10], functions=[partial_first(if ((gid#25 = 0)) first(testtable.`ints`)#29 else null, true), partial_count(if ((gid#25 = 2)) testtable.`doubles`#27 else null), partial_count(if ((gid#25 = 1)) testtable.`longs`#26 else null)])
```

I am wondering if there will be a case where we come across same plan i.e  `if` inside  `partial_first` with `agg` in it. i.e will this code be falling back to CPU even without multi distinct aggs in some cases.

Reply:
First is non-deterministic , is that what you are seeing? Seems that way unless I am not getting something.